File: src/domain/entities/operational/SessionStatus.ts
import { SessionReference } from './SessionReference.js';

export enum SessionStatus {
  Starting = 'starting',
  Stopping = 'stopping',
  Stopped = 'stopped',
  Failed = 'failed',
  Connected = 'connected',
  Reconnecting = 'reconnecting',
  LoggedOut = 'logged_out',
}

/**
 * Operational state transition emitted by the session host/runtime lifecycle.
 */
export class SessionStatusEvent {
  constructor(
    public readonly session: SessionReference,
    public readonly status: SessionStatus,
    public readonly timestamp: string,
    public readonly workerId?: string,
    public readonly reason?: string,
  ) {}
}

File: sdks/java/src/main/java/com/digows/whatsappgateway/operational/SessionStatus.java
package com.digows.whatsappgateway.operational;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonValue;

public enum SessionStatus
{
  STARTING("starting"),
  STOPPING("stopping"),
  STOPPED("stopped"),
  FAILED("failed"),
  CONNECTED("connected"),
  RECONNECTING("reconnecting"),
  LOGGED_OUT("logged_out");

  private final String wireValue;

  SessionStatus(String wireValue)
  {
    this.wireValue = wireValue;
  }

  @JsonValue
  public String getWireValue()
  {
    return wireValue;
  }

  @JsonCreator
  public static SessionStatus fromWireValue(String wireValue)
  {
    for (SessionStatus value : values())
    {
      if (value.wireValue.equals(wireValue))
      {
        return value;
      }
    }

    throw new IllegalArgumentException("Unsupported session status \"" + wireValue + "\".");
  }
}
